tests/service/test_service_without_skip_host_check.yml

```diff
@@ -0,0 +1,520 @@
+      - hostgroup01
+      - hostgroup02
+      action: member
+      state: absent
+    register: result
+    failed_when: not result.changed
+
+  - name: Service "HTTP/{{ svc_fqdn }}" members allow_retrieve_keytab absent for users, groups, hosts and hostgroups, again.
+    ipaservice:
+      ipaadmin_password: SomeADMINpassword
+      name: "HTTP/{{ svc_fqdn }}"
+      allow_retrieve_keytab_user:
+      - user01
+      - user02
+      allow_retrieve_keytab_group:
+      - group01
+      - group02
+      allow_retrieve_keytab_host:
+      - "{{ host1_fqdn }}"
+      - "{{ host2_fqdn }}"
+      allow_retrieve_keytab_hostgroup:
+      - hostgroup01
+      - hostgroup02
+      action: member
+      state: absent
+    register: result
+    failed_when: result.changed
+
+  #
+  - name: Ensure service is absent
+    ipaservice:
+      ipaadmin_password: SomeADMINpassword
+      name: "HTTP/{{ svc_fqdn }}"
+      state: absent
+    register: result
+    failed_when: not result.changed
+
+  - name: Ensure service is absent, again
+    ipaservice:
+      ipaadmin_password: SomeADMINpassword
+      name: "HTTP/{{ svc_fqdn }}"
+      state: absent
+    register: result
+    failed_when: result.changed
+
+  # cleanup
+
+  - name: Ensure services are absent.
+    ipaservice:
+      ipaadmin_password: SomeADMINpassword
+      name:
+      - "HTTP/{{ svc_fqdn }}"
+      - HTTP/svc.ihavenodns.info
+      state: absent
+
+  - name: Ensure host is absent
+    ipahost:
+      ipaadmin_password: SomeADMINpassword
+      name:
+      - "{{ svc_fqdn }}"
+      - "{{ host1_fqdn }}"
+      - "{{ host2_fqdn }}"
+      - svc.ihavenodns.info
+      state: absent
+
+  - name: Ensure testing users are absent.
+    ipauser:
+      ipaadmin_password: SomeADMINpassword
+      name:
+      - user01
+      - user02
+      state: absent
+
+  - name: Ensure testing groups are absent.
+    ipagroup:
+      ipaadmin_password: SomeADMINpassword
+      name:
+      - group01
+      - group02
+      state: absent
+
+  - name: Ensure testing hostgroup hostgroup01 is absent.
+    ipagroup:
+      ipaadmin_password: SomeADMINpassword
+      name:
+      - hostgroup01
+      state: absent
+
+  - name: Ensure testing hostgroup hostgroup02 is absent.
+    ipagroup:
+      ipaadmin_password: SomeADMINpassword
+      name:
+      - hostgroup02
+      state: absent
+
+  # If ansible-freeipa dnsrecord module is availble, it can be used.
+  # If ansible-freeipa dnsrecord module is availble, it can be used.
+  - name: DNS record for "{{ svc_fqdn }}" is not present.
+    command: ipa dnsrecord-del --del-all "{{ ipaserver_domain }}" svc
+    ignore_errors: yes
+
+  - name: DNS record for "{{ host1_fqdn }}" is not present.
+    command: ipa dnsrecord-del --del-all "{{ ipaserver_domain }}" host1
+    ignore_errors: yes
+
+  - name: DNS record for "{{ host2_fqdn }}" is not present.
+    command: ipa dnsrecord-del --del-all "{{ ipaserver_domain }}" host2
+    ignore_errors: yes
```

Comment:
...

Reply:
Better:

    register: result
    failed_when: result is not defined
    changed_when: result is defined and
                  "DNS resource record not found" not in result.stderr

Reply:
Fixed by removing usage of `ipa dnsrecord-*` and using ansible-freeipa host module.